### PR TITLE
DCO + Update Contributing.md

### PR DESCRIPTION
**Pull request by EthanPERRUZZA**

### CODE REVIEW

**Comment by emersion** [comment]
Should we also add a CI check to make sure that no S-o-b lines are missing? (We already have a CI check for the commit titles.)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 
-Check out our community's [Code of Conduct](https://github.com/OpenRailAssociation/osrd/blob/dev/CODE_OF_CONDUCT.md) and feel free to say hi on [libera chat](https://web.libera.chat/#osrd) if you'd like. It's a nice place to chat about OSRD development, ask questions, and get to know the other contributors and users in a less formal setting. If you are not sure about what to do have a look at currently open [good first issues](https://github.com/OpenRailAssociation/osrd/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Check out our community's [Code of Conduct](https://github.com/OpenRailAssociation/osrd/blob/dev/CODE_OF_CONDUCT.md) and feel free to say hi on our [#public-general:osrd.fr](https://matrix.to/#/#public-general:osrd.fr)matrix channel if you'd like. It's a nice place to chat about OSRD development, ask questions, and get to know the other contributors and users in a less formal setting. If you are not sure about what to do have a look at currently open [good first issues](https://github.com/OpenRailAssociation/osrd/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
```
Comment (emersion) [inline CONTRIBUTING.md]: A space is missing between the link and "matrix channel".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,16 +70,17 @@ You can join the OSRD team on [transifex](https://explore.transifex.com/osrd/osr
...
+include a Signed-off-by line. You can find more information about this here:
+[osrd.fr/docs/guides/contribute/contribute-code/commit-conventions/#the-developer-certificate-of-origin](https://osrd.fr/en/docs/guides/contribute/contribute-code/commit-conventions/#the-developer-certificate-of-origin)
+
+For more advice on how to contribute, follow that link:
+[osrd.fr/docs/contribute/contribute-code](https://osrd.fr/en/docs/guides/contribute/contribute-code).
 
 ### Pull request / code review
 
-[osrd.fr/docs/contribute/code/#share-your-changes](https://osrd.fr/en/docs/guides/contribute/code/#share-your-changes).
+[osrd.fr/docs/contribute/contribute-code/share-your-changes](https://osrd.fr/en/docs/guides/contribute/contribute-code/share-your-changes).
```
Comment (emersion) [inline CONTRIBUTING.md]: Note to self, this page is sitting in a PR: https://github.com/OpenRailAssociation/osrd-website/pull/231

**Comment by Khoyo** [comment]
> Should we also add a CI check to make sure that no S-o-b lines are missing? (We already have a CI check for the commit titles.)

I believe we intended to use https://github.com/apps/dco to perform the check

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -70,16 +70,17 @@ You can join the OSRD team on [transifex](https://explore.transifex.com/osrd/osr
 
 ## Contribute code
 
-[osrd.fr/docs/contribute/code](https://osrd.fr/en/docs/guides/contribute/code).
+To comply with the [DCO](http://developercertificate.org/), all commits must
+include a Signed-off-by line. You can find more information about this here:
+[osrd.fr/docs/guides/contribute/contribute-code/commit-conventions/#the-developer-certificate-of-origin](https://osrd.fr/en/docs/guides/contribute/contribute-code/commit-conventions/#the-developer-certificate-of-origin)
```
Comment (flomonster) [inline CONTRIBUTING.md]:
```suggestion
include a Signed-off-by line. You can find more information about this [here](https://osrd.fr/en/docs/guides/contribute/contribute-code/commit-conventions/#the-developer-certificate-of-origin)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -70,16 +70,17 @@ You can join the OSRD team on [transifex](https://explore.transifex.com/osrd/osr
...
+
+For more advice on how to contribute, follow that link:
+[osrd.fr/docs/contribute/contribute-code](https://osrd.fr/en/docs/guides/contribute/contribute-code).
```
Comment (flomonster) [inline CONTRIBUTING.md]:
```suggestion
https://osrd.fr/en/docs/guides/contribute/contribute-code
```

**Review by flomonster — APPROVED** [review]
LGTM. 

I'm just nitpicking that the hyperlinks have the full url as text.

**Comment by EthanPERRUZZA** [comment]
> > Should we also add a CI check to make sure that no S-o-b lines are missing? (We already have a CI check for the commit titles.)
> 
> I believe we intended to use https://github.com/apps/dco to perform the check

In the issue, it is mention that we will only need activate a setting to check if a commit was signed off
https://github.com/osrd-project/osrd-confidential/issues/602#issue-2418565021

**Pull request merged**